test/unit/Cover/buyCover.js

```diff
@@ -802,4 +805,464 @@ describe('buyCover', function () {
       ),
     ).to.be.revertedWithPanic('0x12'); // (Division or modulo division by zero)
   });
+
+  it('reverts if allocationRequest coverAmountInAsset is 0', async function () {
+    const { cover } = this;
+
+    const {
+      members: [coverBuyer],
+    } = this.accounts;
+
+    const { amount, productId, coverAsset, period, expectedPremium } = buyCoverFixture;
+
+    await expect(
+      cover.connect(coverBuyer).buyCover(
+        {
+          owner: coverBuyer.address,
+          productId,
+          coverAsset,
+          amount,
+          period,
+          maxPremiumInAsset: expectedPremium,
+          paymentAsset: coverAsset,
+          payWitNXM: false,
+          commissionRatio: parseEther('0'),
+          commissionDestination: AddressZero,
+          ipfsData: '',
+        },
+        [{ poolId: '0', coverAmountInAsset: 0 }],
+        {
+          value: expectedPremium,
+        },
+      ),
+    ).to.be.revertedWith('Cover: coverAmountInAsset = 0'); // (Division or modulo division by zero)
+  });
+
+  it('retrieves ERC20 payment from caller', async function () {
```

Comment:
let's enhance this test to 

`retrieves ERC20 payment from caller and transfers it to the Pool`

Reply:
Adding the right assertion that the pool DAI balance has increased